src/servers/model/utils/fetch-retry.ts

```diff
@@ -31,3 +34,15 @@ export async function fetchRetry(url: string, timeout: number, retryCount: numbe
 
     return result;
 }
+
+/** Like `fetch` but supports Google Cloud Storage (gs://) protocol. */
+export async function extendedFetch(url: string, init?: fetch.RequestInit): Promise<Response> {
+    if (url.startsWith('gs://')) return fetch_GS(url, init);
+    else return fetch(url, init);
+}
+
+async function fetch_GS(url: string, init?: fetch.RequestInit): Promise<Response> {
+    if (init?.signal?.aborted) throw new AbortError('The user aborted a request.');
+    const data = await downloadGs(url);
+    return new Response(data, init);
+}
```

Comment:
If there is no await, no need for async.

```suggestion
export function wrapFetch(url: string, init?: fetch.RequestInit): Promise<Response> {
    if (url.startsWith('gs://')) return fetchGS(url, init);
    else return fetch(url, init);
}

async function fetchGS(url: string, init?: fetch.RequestInit): Promise<Response> {
    if (init?.signal?.aborted) throw new AbortError('The user aborted a request.');
    const data = await downloadGs(url);
    return new Response(data, init);
}
```

Reply:
I always get a bit confused by functions that are not async but return a promise :D
But you're right.
I changed it.